lang/cs/Org.Apache.REEF.Network/NetworkService/StreamingNetworkService.cs

```diff
@@ -42,37 +39,78 @@ public class StreamingNetworkService<T> : INetworkService<T>
         private IIdentifier _localIdentifier;
-        private readonly IDisposable _messageHandlerDisposable;
+        private readonly IDisposable _messageHandlerDisposable1;
+        private readonly IDisposable _messageHandlerDisposable2;
         private readonly Dictionary<IIdentifier, IConnection<T>> _connectionMap;
         private readonly INameClient _nameClient;
 
         /// <summary>
         /// Create a new Writable NetworkService.
         /// </summary>
         /// <param name="messageHandler">The observer to handle incoming messages</param>
-        /// <param name="idFactory">The factory used to create IIdentifiers</param>
         /// <param name="nameClient">The name client used to register Ids</param>
-        /// <param name="remoteManagerFactory">Writable RemoteManagerFactory to create a 
-        /// Writable RemoteManager</param>
+        /// <param name="remoteManagerFactory">
+        /// Writable RemoteManagerFactory to create a Writable RemoteManager
+        /// </param>
         /// <param name="codec">Codec for Network Service message</param>
         /// <param name="localAddressProvider">The local address provider</param>
-        /// <param name="injector">Fork of the injector that created the Network service</param>
         [Inject]
         private StreamingNetworkService(
             IObserver<NsMessage<T>> messageHandler,
-            IIdentifierFactory idFactory,
             INameClient nameClient,
             StreamingRemoteManagerFactory remoteManagerFactory,
             NsMessageStreamingCodec<T> codec,
-            ILocalAddressProvider localAddressProvider,
-            IInjector injector)
+            ILocalAddressProvider localAddressProvider)
+            : this(messageHandler, null, nameClient, remoteManagerFactory, codec, localAddressProvider)
+        {
+        }
+
+        /// <summary>
+        /// Create a new Writable NetworkService
+        /// </summary>
+        /// <param name="messageHandler">The observer to handle incoming messages</param>
+        /// <param name="nameClient">The name client used to register Ids</param>
+        /// <param name="remoteManagerFactory">
+        /// Writable RemoteManagerFactory to create a Writable RemoteManager
+        /// </param>
+        /// <param name="codec">Codec for Network Service message</param>
+        /// <param name="localAddressProvider">The local address provider</param>
+        [Inject]
+        private StreamingNetworkService(
+            IObserver<IRemoteMessage<NsMessage<T>>> messageHandler,
+            INameClient nameClient,
+            StreamingRemoteManagerFactory remoteManagerFactory,
+            NsMessageStreamingCodec<T> codec,
+            ILocalAddressProvider localAddressProvider)
+            : this(null, messageHandler, nameClient, remoteManagerFactory, codec, localAddressProvider)
+        {
+        }
+
+        [Inject]
+        private StreamingNetworkService(
```

Comment:
They are needed in injecting the object. It is used to handle the scenarios of binding the different combinations of types of universal observers.